linked_list/linked_list.py

```diff
@@ -9,71 +12,150 @@ def __init__(self, value, next_node = None):
 # Defines the singly linked list
 class LinkedList:
     def __init__(self):
-      self.head = None # keep the head private. Not accessible outside this class
+        self.head = None # keep the head private. Not accessible outside this class
 
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def get_first(self):
-        pass
+        if not self.head:
+            return None
+        else:
+            return self.head.value
 
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_first(self, value):
-        pass
+        new_node = Node(value)
+        new_node.next = self.head
+        self.head = new_node
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def search(self, value):
-        pass
+        # get head and iterate through nodes until right value
+        current_node = self.head
+        if not current_node:
+            return False
+        
+        while current_node.value != value:
+            if current_node.next:
+                current_node = current_node.next
+            else:
+                return False
+        else:
+            return True
 
     # method that returns the length of the singly linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def length(self):
-        pass
+        # start counter, go through list, if next increase counter
+        current_node = self.head
+        length = 0
+        while current_node:
+            length += 1
+            if current_node.next:
+                current_node = current_node.next
+            else:
+                return length
+        else:
+            return length

```

Comment:
🤓 Style suggestion to condense your code. Since your while condition is essentially to check if `current_node` is a Node object or `None`, just set `current_node` to `current_node.next` regardless of whether next is a Node or `None`. 

If it's `None`, your while loop will finish executing and you can just return whatever `length` is at that point!
```suggestion
            current_node = current_node.next
        return length

```